src/components/ChannelPreview/hooks/useChannelPreviewDisplayName.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect, useContext } from 'react';
+import { ChatContext } from '../../../context';
+
+export const useChannelPreviewDisplayName = (channel) => {
+  const { client } = useContext(ChatContext);
+  const [displayName, setDisplayName] = useState('');
+
+  useEffect(() => {
+    if (channel.data.name && typeof channel.data.name === 'string') {
+      setDisplayName(channel.data.name);
+      return;
```

Comment:
if the rest of this function only needs to be executed if `channel.data.name` exists throw an `else` in here for the rest of the effect and remove this return as a return from a useEffect is the cleanup function that'll run